pkg/utils/transform/transform.go

```diff
@@ -109,6 +118,151 @@ func InjectLabel(key, value string, overwritePolicy OverwritePolicy, kinds ...st
+
+		return nil
+	}
+}
+
+func replaceContainerImages(containers []corev1.Container, images map[string]string) {
+	for i, container := range containers {
+		name := formKey("", container.Name)
+		if url, exist := images[name]; exist {
+			containers[i].Image = url
+		}
+
+		replaceContainersArgsImage(&container, images)
+	}
+}
+
+func replaceContainersArgsImage(container *corev1.Container, images map[string]string) {
+	for a, arg := range container.Args {
+		if argVal, hasArg := splitsByEqual(arg); hasArg {
+			argument := formKey(ARG_PREFIX, argVal[0])
+			if url, exist := images[argument]; exist {
+				container.Args[a] = argVal[0] + "=" + url
+			}
+			continue
+		}
+
+		argument := formKey(ARG_PREFIX, arg)
+		if url, exist := images[argument]; exist {
+			container.Args[a+1] = url
+		}
+	}
+
+}
+
+func formKey(prefix, arg string) string {
+	argument := strings.ToLower(arg)
+	if prefix != "" {
+		argument = prefix + "_" + argument
+	}
+	argument = strings.ReplaceAll(argument, "-", "_")
+	return argument
+}
+
+func splitsByEqual(arg string) ([]string, bool) {
+	values := strings.Split(arg, "=")
+	if len(values) == 2 {
+		return values, true
+	}
+
+	return values, false
+}
+
+func TaskImages(images map[string]string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "ClusterTask" {
+			return nil
+		}
+
+		steps, found, err := unstructured.NestedSlice(u.Object, "spec", "steps")
+		if err != nil {
+			return err
+		}
+		if !found {
+			return nil
+		}
+		replaceStepsImages(steps, images)
+		err = unstructured.SetNestedField(u.Object, steps, "spec", "steps")
+		if err != nil {
+			return err
+		}
+
+		params, found, err := unstructured.NestedSlice(u.Object, "spec", "params")
+		if err != nil {
+			return err
+		}
+		if !found {
+			return nil
+		}
+		replaceParamsImage(params, images)
+		err = unstructured.SetNestedField(u.Object, params, "spec", "params")
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func replaceStepsImages(steps []interface{}, override map[string]string) {
+	for _, s := range steps {
+		step := s.(map[string]interface{})
+		name, ok := step["name"].(string)
+		if !ok {
+			transformLog.Info(fmt.Sprintf("Unable to get the step name %v", s))
+			continue
+		}
+
+		name = formKey("", name)
+		image, found := override[name]
+		if !found || image == "" {
+			transformLog.Info(fmt.Sprintf("Image not found for %s step, skipping replacement", name))
+			continue
+		}
+		step["image"] = image
+	}
+}
+
+func replaceParamsImage(params []interface{}, override map[string]string) {
+	for _, p := range params {
+		param := p.(map[string]interface{})
+		name, ok := param["name"].(string)
+		if !ok {
+			transformLog.Info(fmt.Sprintf("Unable to get the step name %v", p))
+			continue
+		}
+
+		name = formKey(PARAM_PREFIX, name)
+		image, found := override[name]
+		if !found || image == "" {
+			transformLog.Info(fmt.Sprintf("Image not found for %s param, skipping replacement", name))
```

Comment:
it is better to use multiple fields / structured logging here e.g. `transformLog.Info("Image not found", "param", name, "action",  "skip")`